tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/types/never/NeverTypeTest.java

```diff
@@ -0,0 +1,145 @@
+        BRunUtil.invoke(neverTypeTestResult, "testExclusiveRecord");
+    }
+
+    @Test(description = "Test XML with 'never' type constraint")
+    public void testXMLWithNeverTypeConstraint() {
+        BRunUtil.invoke(neverTypeTestResult, "testXMLWithNeverType");
+    }
+
+    @Test(description = "Test union type with 'never' type: 1")
+    public void testNeverWithUnionType1() {
+        BRunUtil.invoke(neverTypeTestResult, "testNeverWithUnionType1");
+    }
+
+    @Test(description = "Test union type with 'never' type: 2")
+    public void testNeverWithUnionType2() {
+        BRunUtil.invoke(neverTypeTestResult, "testNeverWithUnionType1");
+    }
+
+    @Test(description = "Test union type with 'never' type: 3")
+    public void testNeverWithUnionType3() {
+        BRunUtil.invoke(neverTypeTestResult, "testNeverWithUnionType3");
+    }
+
+    @Test(description = "Test table's key constraint with 'never' type")
+    public void testNeverWithKeyLessTable() {
+        BRunUtil.invoke(neverTypeTestResult, "testNeverWithKeyLessTable");
+    }
+
+    @Test(description = "Test table key constraint with 'never' type")
+    public void testNeverInUnionTypedKeyConstraints() {
+        BRunUtil.invoke(neverTypeTestResult, "testNeverInUnionTypedKeyConstraints");
+    }
+
+    @Test(description = "Test 'never' type as future type param")
+    public void testNeverAsFutureTypeParam() {
+        BRunUtil.invoke(neverTypeTestResult, "testNeverAsFutureTypeParam");
+    }
+
+    @Test(description = "Test 'never' type as mapping type param")
+    public void testNeverAsMappingTypeParam() {
+        BRunUtil.invoke(neverTypeTestResult, "testNeverAsMappingTypeParam");
+    }
+
+    @Test
+    public void testNeverTypeNegative() {
+        Assert.assertEquals(negativeCompileResult.getErrorCount(), 19);
+        int i = 0;
+        BAssertUtil.validateError(negativeCompileResult, i++,
+                "'never'-typed variable definitions not allowed", 2, 5);
+        BAssertUtil.validateError(negativeCompileResult, i++,
+                "'never'-typed variable definitions not allowed", 12, 5);
+        BAssertUtil.validateError(negativeCompileResult, i++,
+                "incompatible types: expected '()', found 'never'", 16, 12);
+        BAssertUtil.validateError(negativeCompileResult, i++,
+                "incompatible types: expected 'never', found 'string'", 25, 12);
+        BAssertUtil.validateError(negativeCompileResult, i++,
+                "incompatible types: expected 'never', found 'string'", 31, 16);
+        BAssertUtil.validateError(negativeCompileResult, i++,
+                "incompatible types: expected 'never', found '()'", 36, 12);
+        BAssertUtil.validateError(negativeCompileResult, i++,
+                "incompatible types: expected 'never', found '()'", 41, 20);
+        BAssertUtil.validateError(negativeCompileResult, i++,
+                "never typed required fields not allowed, define 'y' as an optional field instead", 49, 5);
```

Comment:
The error message should've been 
```suggestion
                "'never' typed required fields not allowed, define 'y' as an optional field instead", 49, 5);
```
right?

Reply:
Yeah. :grimacing: My bad